qa/workunits/rbd/cli_generic.sh

```diff
@@ -27,6 +27,37 @@ remove_images() {
     done
 }
 
+test_diff() {
+    local pool_name=xrbddiff1
+    local image_name=xtestdiff1
+    local snapshot_name=snap1
+    local image_size=1M
+    ceph osd pool create $pool_name 8
+    rbd create --thick-provision --size $image_size $pool_name/$image_name
+    test -n "$(rbd diff $pool_name/$image_name)"
+    rbd rm $pool_name/$image_name
+    rbd create --size $image_size $pool_name/$image_name
+    test -z "$(rbd diff $pool_name/$image_name)"
+    rbd snap create $pool_name/$image_name --snap=allzeroes
+    test -z "$(rbd diff $pool_name/$image_name)"
+    test -z "$(rbd diff --from-snap=allzeroes $pool_name/$image_name)"
+    local device="$(sudo rbd map $pool_name/$image_name)" # /dev/rbdx
```

Comment:
Instead of mapping the device (which depends on sudo and krbd) you could write to the image e.g. with rbd bench:
```
rbd bench --io-type write --io-size 1M --io-total 1M $POOL_NAME/$IMAGE_NAME
```

Reply:
> Instead of mapping the device (which depends on sudo and krbd) you could write to the image e.g. with rbd bench:
> 
> ```
> rbd bench --io-type write --io-size 1M --io-total 1M $POOL_NAME/$IMAGE_NAME
> ```

is there a possibility to write to an image by particular offset?

Reply:
No. In the above example it will start from offset zero, because by default `rbd bench` runs sequential writes and the first thread starts from the offset 0. If you need to write more data to a large image then make sure to specify '--io-threads 1`.